<launch>
  <arg name="map" value="icra2019"/>
  <arg name="world" value="1vs1"/>
  <master auto="start"/>
  <param name="/use_sim_time" value="true"/>




  <!-- Run the stage simulator -->
  <node pkg="stage_ros" type="stageros" name="stageros" args="$(find icra_roboin_simulation)/worlds/$(arg world).world" respawn="false" >
    <remap from="robot_0/base_scan" to="robot_0/scan" />
    <remap from="robot_1/base_scan" to="robot_1/scan" />
    <param name="base_watchdog_timeout" value="0.01" />
  </node>


  <group ns="robot_0">
	  <param name="use_sim_time" value="true"/>
   	  <!--param name="tf_prefix" value="robot_0" /-->

	  <!-- Run the map server -->
	  <node name="map_server" pkg="map_server" type="map_server" args="$(find icra_roboin_simulation)/maps/$(arg map).yaml" respawn="false" />

  	  <!--Load parameters for localization node  -->
	  <rosparam command="load" file="$(find roborts_localization)/config/localization.yaml" />
  	  <rosparam command="load" file="$(find roborts_localization)/amcl/config/amcl.yaml" />

	  <!-- Run the localization node -->
	  <node pkg="roborts_localization" type="localization_node" name="localization_node" respawn="false" />

	  <include file="$(find icra_roboin_navigation)/launch/navigation_base.launch"/>

	  <!-- detection simulator node -->
	  <node pkg="icra_roboin_simulation" type="stage_blob_detector.py" name="yolo_simulator"/>

	  <!-- behavior interface node -->
	  <node pkg="icra_roboin_behavior" type="behavior_node" name="behavior_node" respawn="false" output="screen"/>

	  </group>

  <!--group ns="robot_1">
	  <param name="~tf_prefix" value="robot_1" />


	  
	  <node name="map_server" pkg="map_server" type="map_server" args="$(find icra_roboin_simulation)/maps/$(arg map).yaml" respawn="false" />

	  
	  <node pkg="roborts_localization" type="localization_node" name="localization_node" respawn="false" />

	  <include file="$(find icra_roboin_navigation)/launch/navigation_base.launch"/>

	  
	  <node pkg="icra_roboin_simulation" type="stage_blob_detector.py" name="yolo_simulator"/>

	  
	  <node pkg="icra_roboin_behavior" type="behavior_node" name="behavior_node" respawn="false" output="screen"/>

  </group-->


  <!-- Run the rviz -->
  <node name="rviz" pkg="rviz" type="rviz" args="-d $(find icra_roboin_simulation)/rviz/1vs1.rviz" />


</launch>
